src/components/confetti/Confetti.tsx

```diff
@@ -0,0 +1,56 @@
+import ReactConfetti from "react-confetti";
+import React, { useEffect } from "react";
+import { styled } from "@storybook/theming";
+import { createPortal } from "react-dom";
+
+interface ConfettiProps
+  extends Omit<React.ComponentProps<typeof ReactConfetti>, "drawShape"> {
+  top: number;
+  left: number;
+  width: number;
+  height: number;
+}
+
+const confettiContainer = document.createElement("div");
```

Comment:
shouldn't this happen dynamically instead of being defined at the top level of the file? This could cause unintended side effects

Reply:
Attaching to the DOM happens dynamically in the useEffect. Also its removal. Creating the div can be a one-time thing

Reply:
good point, but this could be done inside of the use effect as well

Reply:
Why should we create every time a fresh div container? We could, be is it necessary? React Portal will unmount the component from the div container, as soon as it is not mounted in the React application. There shouldn't be any side-effects.